src/journal/journald-native.c

```diff
@@ -81,60 +81,48 @@ static bool allow_object_pid(const struct ucred *ucred) {
         return ucred && ucred->uid == 0;
 }
 
-void server_process_native_message(
+static int server_process_entry(
                 Server *s,
-                const void *buffer, size_t buffer_size,
+                const void *buffer, size_t *remaining,
                 const struct ucred *ucred,
                 const struct timeval *tv,
                 const char *label, size_t label_len) {
 
+        /* Process a single entry from a native message.
+         * Returns 0 if nothing special happened and the message processing should continue,
+         * and a negative or positive value otherwise. */
```

Comment:
not a fan of functions that modify their output parameters on failures too. Now, in this case it's probably OK, given that this function is static, but I think this really deserves a comment at least. Something like this: "Note that *remaining is altered on both success and failure" or so